src/Orleans.Serialization/Cloning/IDeepCopier.cs

```diff
@@ -274,8 +274,10 @@ internal static class ShallowCopyableTypes
         {
             [typeof(decimal)] = true,
             [typeof(DateTime)] = true,
+#if NET6_0_OR_GREATER
```

Comment:
These conditions aren't going to work as expected on .NET 5 or 6 because we don't have a build target for these frameworks and so these conditions will be always false when running on .net3.1 - 6.0.